ios/RNCameraRoll.m

```diff
@@ -40,15 +62,18 @@ @implementation RNCameraRoll
       onlyImagesOptions.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"modificationDate" ascending:YES]];
 
       PHFetchResult *allPhotosResult = [PHAsset fetchAssetsWithMediaType:PHAssetMediaTypeImage options:onlyImagesOptions];
+
+      NSMutableArray<AssetData*> *assetDatas = [[NSMutableArray alloc] init];
+
+      dispatch_group_t assetDatasGroup = dispatch_group_create();
```

Comment:
Use a dispatch group to block the thread until all async calls to `_writeToDisk` have completed.